app/src/main/java/tech/relaycorp/gateway/domain/courier/ProcessCargo.kt

```diff
@@ -1,10 +1,26 @@
 package tech.relaycorp.gateway.domain.courier
 
+import tech.relaycorp.gateway.data.disk.CargoStorage
+import tech.relaycorp.gateway.domain.StoreParcel
+import tech.relaycorp.relaynet.cogrpc.readBytesAndClose
+import tech.relaycorp.relaynet.messages.Cargo
 import javax.inject.Inject
 
 class ProcessCargo
-@Inject constructor() {
+@Inject constructor(
+    private val cargoStorage: CargoStorage,
+    private val readParcelsFromCargo: ReadParcelsFromCargo,
+    private val storeParcel: StoreParcel
+) {
 
-    // TODO: implement
-    suspend fun process() {}
+    suspend fun process() {
+        val cargoes = cargoStorage.list()
+        cargoes.forEach { cargoStream ->
+            val cargo = Cargo.deserialize(cargoStream().readBytesAndClose())
+            readParcelsFromCargo
+                .read(cargo)
+                .forEach { storeParcel.store(it) }
```

Comment:
Keep in mind a cargo may contain other things besides parcels, such as PCAs. Happy to keep PCAs outside the scope of this PR though, I just wanted to remind you just in case.

Reply:
Right, forgot about that. I'll add a `TODO` about it.